sync/metrics.go

```diff
@@ -11,40 +12,152 @@ import (
 var meter = otel.Meter("header/sync")
 
 type metrics struct {
-	totalSynced      atomic.Int64
-	totalSyncedGauge metric.Float64ObservableGauge
+	totalSynced     atomic.Int64
```

Comment:
can we add some comment here about how totalSynced differs from subjectiveHead ? It will be confusing for node runners otherwise. We can add as description to otel description of the metric as well. Just make it obvious what it's counting here.

Reply:
added the description